added test to verify ResponseProcessor class.

Sandilya1795 opened the pull request:
Mike, I could push the homework to github. hope to hear from you where am i going wrong for my exchange being returned as null before sending to ResponseProcessor class. Thank you.

Sandilya1795 committed:
added test to verify ResponseProcessor class.

Technolords commented on src/test/java/net/technolords/micro/processor/ResponseProcessorTest.java:
```diff
@@ -0,0 +1,61 @@
+
+    String PATH_TO_CONFIG_FILE = "D:/commit2Learn/data/mockConfigurations/";
+    String PATH_TO_REQUEST = "D:/commit2Learn/data/mockRequests/";
+    String PATH_TO_RESPONSE = "D:/commit2Learn/data/mockResponses/";
```
These path references are pointing to your local environment. As result, when I execute the test, or even do a

```
mvn clean install
```

it fails with:

```
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.38 sec <<< FAILURE! - in TestSuite
testPostResponses(net.technolords.micro.processor.ResponseProcessorTest)  Time elapsed: 0.008 sec  <<< FAILURE!
java.nio.file.NoSuchFileException: D:/commit2Learn/data/mockRequests/sample-post1-request.xml
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
        at java.nio.file.Files.newByteChannel(Files.java:361)
        at java.nio.file.Files.newByteChannel(Files.java:407)
        at java.nio.file.Files.readAllBytes(Files.java:3152)
        at net.technolords.micro.processor.ResponseProcessorTest.testPostResponses(ResponseProcessorTest.java:39)
```

In other words, you'd need to create a config file in src/test/resources just like you're doing with the test resources itself.

Technolords commented on src/test/java/net/technolords/micro/processor/ResponseProcessorTest.java:
```diff
@@ -0,0 +1,61 @@
+
+        Exchange exchange = new DefaultExchange(new DefaultCamelContext());
+        exchange.getIn().setHeader(Exchange.HTTP_METHOD, "POST");
+        exchange.getIn().setHeader(Exchange.HTTP_URI, "/mock/post");
```
Suggestion is to put 'POST' and '/mock/post' as additional parameters? That way you can have a universal test method supporting all scenario's and allows for different test suites?

Technolords reviewed (changes requested):
Unable to reproduce your problem, because I am hitting an error prior to your situation. Please fix the configuration location issue first.

Sandilya1795 committed:
added test to verify ResponseProcessor class.

Sandilya1795 committed:
added test data to verify ResponseProcessor class.

Sandilya1795 committed:
Merge branch 'master' into Homework

Sandilya1795 committed:
- Test Support is moved to Test Classes
- Project POM is corrected with latest version of TestNG and extra dependency is removed.

Technolords committed:
Removed the dot in the end, as it does not match/work
